### PR TITLE
Biqian/coordinator

### PR DESCRIPTION
**Pull request by bjqian**
add management for signalr, aks and kubernetes

### CODE REVIEW

File: src/Pods/Coordinator/ArmDepoyHelper.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Coordinator
+{
+    class ArmDeployHelper
+    {
+        public IAzure GetAzure(AzureEnvironment azureEnvironment)
+        {
+            if (azureEnvironment.Name == "PPE")
+            {
+                return Microsoft.Azure.Management.Fluent.Azure.Configure().Authenticate(PerfConfig.PPE.SERVICE_PRINCIPAL).WithSubscription(PerfConfig.PPE.SUBSCRIPTION);
```
Comment (vwxyzh) [inline src/Pods/Coordinator/ArmDepoyHelper.cs:14]: add `using FluentAzure = Microsoft.Azure.Management.Fluent.Azure;`

File: src/Pods/Coordinator/PerfConfig.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Coordinator
+{
+    class PerfConfig
+    {
+        public static string PREFIX_PERF { get; private set; }
```
Comment (vwxyzh) [inline src/Pods/Coordinator/PerfConfig.cs:13]: naming

File: src/Pods/Coordinator/PerfConfig.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Coordinator
+{
+    class PerfConfig
```
Comment (vwxyzh) [inline src/Pods/Coordinator/PerfConfig.cs:11]: visibility

File: src/Pods/Coordinator/PerfConfig.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                );
+            }
+            catch (Exception e)
+            {
+                Environment.Exit(1);
+            }
+            //init ppe
+        }
+
+        private class SP
```
Comment (vwxyzh) [inline src/Pods/Coordinator/PerfConfig.cs:101]: naming

File: src/Pods/Coordinator/ArmDepoyHelper.cs (hunk 1 above)
Comment (bjqian) [reply src/Pods/Coordinator/ArmDepoyHelper.cs]: fixed

File: src/Pods/Coordinator/PerfConfig.cs (hunk 2 above)
Comment (bjqian) [reply src/Pods/Coordinator/PerfConfig.cs]: fixed

File: src/Pods/Coordinator/PerfConfig.cs (hunk 3 above)
Comment (bjqian) [reply src/Pods/Coordinator/PerfConfig.cs]: fixed

File: src/Pods/Coordinator/K8s/AksHelper.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+using Microsoft.Azure.Management.ContainerService.Fluent;
+using Microsoft.Azure.Management.ContainerService.Fluent.Models;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using System.Threading.Tasks;
+
+namespace Coordinator
+{
+    internal class AksHelper
+    {
+        private IAgentPoolsOperations agentPoolsOperations;
```
Comment (vwxyzh) [inline src/Pods/Coordinator/K8s/AksHelper.cs:13]: add `readonly`

File: src/Pods/Coordinator/K8s/KubeCtlHelper.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+{
+    internal class KubeCtlHelper
+    {
+        private Kubernetes kubernetes;
+
+        public KubeCtlHelper()
+        {
+            kubernetes = getKubeClient();
+        }
+        public Kubernetes getKubeClient()
```
Comment (vwxyzh) [inline src/Pods/Coordinator/K8s/KubeCtlHelper.cs:18]: naming

File: src/Pods/Coordinator/K8s/AksHelper.cs (hunk 5 above)
Comment (vwxyzh) [reply src/Pods/Coordinator/K8s/AksHelper.cs:13]: add prefix `_` for fields.

File: src/Pods/Coordinator/PerfConfig.cs [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                );
+            }
+            catch (Exception e)
+            {
+                Environment.Exit(1);
+            }
+            //init ppe
+        }
+
+        private class Sp
```
Comment (vwxyzh) [inline src/Pods/Coordinator/PerfConfig.cs:104]: naming

File: src/Pods/Coordinator/PerfConfig.cs [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+                ServicePrincipal = SdkContext.AzureCredentialsFactory.FromServicePrincipal(
+                    sp.appId,
+                    sp.password,
+                    sp.tenant, AzureEnvironment
+                );
+            }
+            catch (Exception e)
+            {
+                Environment.Exit(1);
```
Comment (vwxyzh) [inline src/Pods/Coordinator/PerfConfig.cs:99]: add log

File: src/Pods/Coordinator/Program.cs [hunk 9]
```diff
@@ -1,15 +1,37 @@
...
+    internal class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+            CreateHostBuilder(args).Build().Run();
         }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
```
Comment (vwxyzh) [inline src/Pods/Coordinator/Program.cs:22]: config logging

File: src/Pods/Coordinator/PerfConfig.cs (hunk 7 above)
Comment (bjqian) [reply src/Pods/Coordinator/PerfConfig.cs:105]: Sp is already Pascal

File: src/Pods/Coordinator/PerfConfig.cs [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+        public static string Subscription { get; private set; }
+
+        public static AzureCredentials ServicePrincipal { get; private set; }
+
+        public static string KubeConfig { get; private set; }
+
+        public static class Queue
+        {
+            public static string PortalJob = "portal-job";
```
Comment (vwxyzh) [inline src/Pods/Coordinator/PerfConfig.cs:44]: `readonly`

File: src/Pods/Coordinator/Program.cs [hunk 11]
```diff
@@ -1,15 +1,43 @@
...
+            .ConfigureLogging((ILoggingBuilder logging) =>
```
Comment (vwxyzh) [inline src/Pods/Coordinator/Program.cs:24]: indent

File: src/Pods/Coordinator/SignalR/SignalRHelper.cs [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Management.SignalR;
+
+namespace Coordinator.SignalR
+{
+    internal class SignalRHelper
+    {
+        private readonly ISignalROperations _signalROperations;
+        private readonly ISignalROperations _signalRPPEOperations;
```
Comment (vwxyzh) [inline src/Pods/Coordinator/SignalR/SignalRHelper.cs:11]: remove

File: src/Pods/Coordinator/SignalR/SignalRHelper.cs [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+        }
+
+        private ISignalROperations GetSignalROperations()
+        {
+            var signalrManagementClient = new SignalRManagementClient(PerfConfig.ServicePrincipal);
+            signalrManagementClient.SubscriptionId = PerfConfig.Subscription;
+            return signalrManagementClient.SignalR;
+        }
+
+        private ISignalROperations GetSignalRPPEOperations()
```
Comment (vwxyzh) [inline src/Pods/Coordinator/SignalR/SignalRHelper.cs:27]: remove

**Pull request merged**